src/backend/decayrange.h

```diff
@@ -0,0 +1,290 @@
+#pragma once
+
+#include "../ds/ptrwrap.h"
+#include "../pal/pal_ds.h"
+#include "largebuddyrange.h"
+
+namespace snmalloc
+{
+  template<typename Rep>
+  class PagemapList
```

Comment:
There's nothing `Pagemap` specific about this (or the stack below), except when in combination with the `Rep`, yes?